src/realm/sync/client.cpp

```diff
@@ -1600,8 +1648,10 @@ void SessionWrapper::actualize(ServerEndpoint endpoint)
         }
     }
 
-    if (!m_client_reset_config)
-        report_progress(); // Throws
+    if (!m_client_reset_config) {
+        // FIXME looks like noop, could be removed without an issue?
+        on_upload_progress(/* only_if_new_uploadable_data = */ true); // Throws
```

Comment:
I would keep it for now (so we definitely have the same behavior) and remove the comment